fix(fix: handle NoneType for half_day_hours in get_number_of_leave_da…

ahmedusama250 opened the pull request:
…ys): Ensure half_day_hours defaults to 4 when None to prevent TypeError during leave calculations.

This commit resolves a TypeError in the get_number_of_leave_days function by ensuring that half_day_hours defaults to 4 when it is None. This prevents the error that occurs when attempting to convert a NoneType to an integer.

Changes:
- Updated half_day_hours assignment to default to 4 if it is None.
- Adjusted number_of_days calculation accordingly.

This fix ensures that leave calculations function correctly even when half_day_hours is not explicitly set.

Closes TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType' in get_number_of_leave_days function #1

Commit:
fix(fix: handle NoneType for half_day_hours in get_number_of_leave_days): Ensure half_day_hours defaults to 4 when None to prevent TypeError during leave calculations.

This commit resolves a TypeError in the get_number_of_leave_days function
by ensuring that half_day_hours defaults to 4 when it is None. This
prevents the error that occurs when attempting to convert a NoneType
to an integer.

Changes:
- Updated half_day_hours assignment to default to 4 if it is None.
- Adjusted number_of_days calculation accordingly.

This fix ensures that leave calculations function correctly even when
half_day_hours is not explicitly set.

Closes TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType' in get_number_of_leave_days function #1

ahmedusama250 commented:
Additionally i changed the indent to 4 spaces

kittiu commented:
> Additionally i changed the indent to 4 spaces

Can you split your bug fix commit and the 4 spaces commit? So the bux fix can be reviewed easily.

Commit:
Revert "/"

This reverts commit 72c7faa498482b0cb5112b4a289c43ab69b6b8a2.

Commit:
fix(fix: handle NoneType for half_day_hours in get_number_of_leave_days): Ensure half_day_hours defaults to 4 when None to prevent TypeError during leave calculations.

This commit resolves a TypeError in the get_number_of_leave_days function
by ensuring that half_day_hours defaults to 4 when it is None. This
prevents the error that occurs when attempting to convert a NoneType
to an integer.

Changes:
- Updated half_day_hours assignment to default to 4 if it is None.
- Adjusted number_of_days calculation accordingly.

This fix ensures that leave calculations function correctly even when
half_day_hours is not explicitly set.

Closes TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType' in get_number_of_leave_days function #1

Commit:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	thai_leave/custom/leave_application.py

ahmedusama250 committed:
Update leave_application.py

fix indent of lines and add miss casting for "half_day_hours"

kittiu commented:
Thanks, ok, I think this make sense.
